examples/custom_generic_nl_family_custom_types.rs

```diff
@@ -0,0 +1,138 @@
+///
+/// THIS EXAMPLE WILL NOT WORK ON YOUR SYSTEM UNLESS YOU CODE AND LOAD A CUSTOM KERNEL MODULE
+/// THAT IMPLEMENTS THIS BASIC (GENERIC) NETLINK FUNCTIONALITY. SEE ALSO HERE:
+/// https://github.com/phip1611/generic-netlink-user-kernel-rust
+fn main() {
+    // this gets registered by your custom Linux kernel module/driver
+    let family_name = "your_family_name_here";
+    // connect
+    let mut sock = NlSocketHandle::connect(
+        NlFamily::Generic,
+        // target pid is 0 => kernel (we send messages to kernel)
+        Some(0),
+        U32Bitmask::empty(),
+    )
+    .unwrap();
+    // family_name to family_id (temporarily assigned netlink family Id by kernel, after registering the generic netlink family)
+    let family_id = sock
+        .resolve_genl_family(family_name)
+        .expect("Kernel Module that registered custom family must exist!");
+    println!(
+        "Netlink family id of generic netlink family with name '{}' is: {}",
+        family_name, family_id
+    );
+
+    // We want to send an Echo command
+    // 1) prepare Foobar Attribute
+    let mut attrs: GenlBuffer<FoobarAttribute, Buffer> = GenlBuffer::new();
+    attrs.push(
+        Nlattr::new(
+            None,
+            false,
+            false,
+            FoobarAttribute::Msg,
+            "Some data that has `Nl` trait implemented, like &str",
+        )
+        .unwrap(),
+    );
+    // 2) prepare Generic Netlink Header. The Generic Netlink Header contains the
+    //    attributes (actual data) as payload.
+    let gnmsghdr = Genlmsghdr::new(
+        FoobarOperation::Echo,
+        // Version is dependent on your kernel module
+        // you can check it there and make special action, or you ignore it
+        1,
+        attrs,
+    );
+    // 3) Prepare Netlink header. The Netlink header contains the Generic Netlink header
+    //    as payload.
+    let nlmsghdr = Nlmsghdr::new(
+        None,
+        family_id,
+        // This depends on your kernel module. Do you check there if any flags are used?
+        // Request is required (TODO by neli or by netlink?)
+        // others are up to you
+        NlmFFlags::new(&[NlmF::Request]),
+        None,
+        // Port ID. Not necessarily the process id of the current process. This is not
+        // used for routing/addressing a socket. You can use this in a application-specific
+        // manner, e.g. to identify a specific socket if your process has multiple sockets.
+        Some(process::id()),
+        NlPayload::Payload(gnmsghdr),
+    );
+
+    // Send data
+    sock.send(nlmsghdr).expect("Send must work");
+
+    // receive echo'ed message
+    let res: Nlmsghdr<u16, Genlmsghdr<FoobarOperation, FoobarAttribute>> =
+        sock.recv().expect("Should receive a message").unwrap();
+
+    // According to Netlink Linux kernel code nl_type is either 0x2 (error) or our family id (ok)
+    if res.nl_type == u16::from(Nlmsg::Error)
```

Comment:
neli has special handling for the `Nlmsg::Error` value as of 0.5.0. This value indicates either an actual error if the error code is non-zero or an ACK if the error code is 0. neli will return these as values of either `NlPayload::Ack` or `NlPayload::Err` in the resulting `Nlmsghdr`. In the error case, `recv()` will detect this as an error and return it as an `Err(Nlmsgerr)` value. Can you explain a little bit more why you chose to do the checking manually here? Does this use case not work for you?

Reply:
Oh, apparently I was using the neli lib wrong in my uni project and therefore also in this demo. 
I didn't had a proper example at the first place that fit my use case, therefore I just tried to figure everything out by myself. I'll improve the code and let you know if it works better. @jbaublitz